src/UiPath.Workflow/Validation/CSharpExpressionValidator.cs

```diff
@@ -75,4 +85,60 @@ protected override string CreateReferenceCode(string types, string names, string
         GetSyntaxTreeForExpression(expressionText);
 
     protected override string GetTypeName(Type type) => CompilerHelper.GetTypeName(type);
+
+    /// <summary>
+    /// Checks if there are multiline comments, and if they are properly closed
+    /// Checks if there are any multiline text statements, and if they are properly closed
+    /// </summary>
+    /// <param name="code"></param>
+    /// <returns></returns>
+    private static bool AreCommentsBalanced(string code)
+    {
+        bool multilineCommentFaulted = false;
+        bool multilineTextFaulted = false;
+        bool isInsideQuotes = false;
+
+        for (int i = 0; i < code.Length - 1; i++)
+        {
+            if (code[i] == '@' && !isInsideQuotes)
+            {
+                multilineTextFaulted = true;
+            }
+            else if (code[i] == '"')
```

Comment:
This doesn't account for escape sequences. If you have a `\"` in your string, `isInsideQuotes` will become `false` and using `@` or `/*` after that point will result in false positives.

Reply:
damn, nice catch!